gamemodes/terrortown/gamemode/server/sv_roleselection.lua

```diff
@@ -526,17 +531,40 @@ local function SetSubRoles(plys, availableRoles, selectableRoles)
 	end
 end
 
+---
+-- Ensure, that baseroles of hardforced subroles are accounted for (eg. received by ULX etc).
+--
+-- @return table List with a count of forced @{ROLE}s
+-- @realm server
+-- @internal
+local function GetHardForcedBaseRoles()
+	local selectedForcedRoles = {}
+
+	for ply, subrole in pairs(roleselection.finalRoles) do
+		if IsValid(ply) then
+			local curRole = roles.GetByIndex(subrole)
+			local isBaseRole = curRole:IsBaseRole()
+
+			-- assign amount of forced players per baserole if this is only a subrole
+			if not isBaseRole then
+				local baserole = curRole.baserole
+				selectedForcedRoles[baserole] = (selectedForcedRoles[baserole] or 0) + 1
+			end
+		end
+	end
+	return selectedForcedRoles
+end
```

Comment:
This now puts baseroles of hardforced subroles into the selectedForcedRoles-List.